test/intl402/NumberFormat/prototype/format/value-tonumber.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018 Ujjwal Sharma. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-number-format-functions
+description: >
+  Tests that Intl.NumberFormat.prototype.format converts its argument
+  (called value) to a number using ToNumber (7.1.3).
+info: |
+  11.1.4Number Format Functions
+
+  4. Let x be ? ToNumber(value).
+features: [Symbol]
+---*/
+
+const toNumberResults = [
+  [undefined, NaN],
+  [null, +0],
+  [true, 1],
+  [false, 0],
+  ['42', 42],
+  ['foo', NaN]
+];
+
+const nf = new Intl.NumberFormat();
+
+toNumberResults.forEach(pair => {
+  const [value, result] = pair;
+  assert.sameValue(nf.format(value), nf.format(result));
+});
+
+let count = 0;
+const dummy = {};
+dummy[Symbol.toPrimitive] = hint => (hint === 'number' ? ++count : NaN);
```

Comment:
@@toPrimitive takes two args: input, hint

Reply:
@gsathya I was referring to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive, which only took one argument. Isn't the input stored at `this` anyway?